lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -97,6 +102,26 @@ def _on_relation_changed(self, event) -> None:
                 },
             )
 
+    def _on_relation_broken(self, event) -> None:
+        # Only relation_deparated events can check if scaling down
+        departed_relation_id = event.relation.id
+        if not self.charm.has_departed_run(departed_relation_id):
+            logger.info(
+                "Deferring, must wait for relation departed hook to decide if relation should be removed."
+            )
+            event.defer()
+            return
+
+        if not self.pass_hook_checks(event):
+            logger.info("Skipping relation broken event: hook checks did not pass")
+            return
```

Comment:
why we are not deferring event here?

Reply:
please read the function `pass_hook_checks `, it defers when necessary